packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -1960,6 +1967,11 @@ function runFormStateAction<S, P>(
   if (__DEV__) {
     ReactCurrentBatchConfig.transition._updatedFibers = new Set();
   }
+
+  // Optimistically update the pending state, similar to useTransition.
+  // This will be reverted automatically when all actions are finished.
+  setPendingState(true);
```

Comment:
Optimistic updates are OP

@acdlite please review the implementation because even though tests pass, I'm sure I screwed something up